### PR TITLE
v1.0.0

### PR DESCRIPTION
**Pull request by betzrhodes**
initial release

### CODE REVIEW

File: BQ25895M.device.lib.nut [hunk 1]
```diff
@@ -89,7 +89,7 @@ class BQ25895M {
     //PUBLIC METHODS
     
     // Initialize battery charger with standard configuration
-    function configureCharger(voltage = null, current = null) {
+    function enableCharger(voltage = null, current = null, settings = {}) {
```
Comment (betzrhodes) [inline BQ25895M.device.lib.nut]: Voltage and current should be required parameters.

File: BQ25895M.device.lib.nut [hunk 2]
```diff
@@ -102,29 +102,27 @@ class BQ25895M {
...
 
         // Set default charge current limit to 1A 
         // Note: Register default is 2048mA, need to 
         // start watchdog to keep any other setting
-        setChargeCurrent(1000); 
+        if(current == null) current = 1000;
+        _setChargeCurrent(current); 
 
+        if(settings == {}){ 
+            _setChargeCurrentOptimizer();
```
Comment (betzrhodes) [inline BQ25895M.device.lib.nut]: If no settings are passed in nothing should be set. 

The setChargeCurrentOptimizer method should only be called if it is in the settings table. 

The code should look something like this: 

if ("chargeCurrentOptimizer" in settings) {
    _setChargeCurrentOptimizer(settings.chargeCurrentOptimizer);
}

_setChargeCurrentOptimizer should be updated to take a boolean "true" or "false", whether this setting is enabled or disabled.

File: BQ25895M.device.lib.nut [hunk 3]
```diff
@@ -102,29 +102,27 @@ class BQ25895M {
...
-    }
-    
-    // Set the enable charging bit, charging will happen automatically
-    function enableCharging() {
         
-        local rd = _getReg(BQ25895M_REG03);
-        rd = rd | (1 << 4); // set CHG_CONFIG bit
-        
-        _setReg(BQ25895M_REG03, rd);
+        _enableCharging();
```
Comment (betzrhodes) [inline BQ25895M.device.lib.nut]: Is this redundant? Isn't charging enabled on line 99 when REG 3 is written to.

File: BQ25895M.device.lib.nut [hunk 4]
```diff
@@ -291,11 +241,88 @@ class BQ25895M {
     
     //-------------------- PRIVATE METHODS --------------------//
     
+    // Set the enable charging bit, charging will happen automatically
+    function _enableCharging() {
```
Comment (betzrhodes) [inline BQ25895M.device.lib.nut]: I don't think this method is needed anymore - see comments in enableCharger mehtod

File: BQ25895M.device.lib.nut [hunk 5]
```diff
@@ -291,11 +241,88 @@ class BQ25895M {
...
+            vreg = 4608; 
+        }
+        
+        local rd = _getReg(BQ25895M_REG06);
+        rd = rd & ~(0xFC); // clear current limit bits
+        rd = rd | (0xFC & (((vreg - 3840) / 16).tointeger()) << 2); // 3840mV is the default offset, 16mV is the resolution
+
+        _setReg(BQ25895M_REG06, rd);
+
+        // Note: Register default is 4.352V, need to
```
Comment (betzrhodes) [inline BQ25895M.device.lib.nut]: This can be removed since watchdog is set in enableCharger method

File: BQ25895M.device.lib.nut [hunk 6]
```diff
@@ -291,11 +241,88 @@ class BQ25895M {
...
+            sichg = 5056;
+        }
+        
+        local rd = _getReg(BQ25895M_REG04);
+        rd = rd & ~(0x7F); // clear current limit bits
+        rd = rd | (0x7F & ichg / 64); // 64mA is the resolution
+    
+        _setReg(BQ25895M_REG04, rd);
+        
+        // Note: Register default is 2048mA, need to
```
Comment (betzrhodes) [inline BQ25895M.device.lib.nut]: This can be removed since it is called in enableCharger method

File: BQ25895M.device.lib.nut [hunk 7]
```diff
@@ -291,11 +241,88 @@ class BQ25895M {
...
+    
+        _setReg(BQ25895M_REG04, rd);
+        
+        // Note: Register default is 2048mA, need to 
+        // start watchdog to keep any other setting
+        _kickWatchdog();
+        
+    }
+    
+    function _setChargeCurrentOptimizer() {
```
Comment (betzrhodes) [inline BQ25895M.device.lib.nut]: _setChargeCurrentOptimizer should be updated to take a boolean "true" or "false", so this setting can be enabled or disabled.

File: README.md [hunk 8]
```diff
@@ -32,7 +33,7 @@ batteryCharger <- BQ25895M(i2c);
   
 ## Class Methods ##
 
-### configureCharger(*[chargeVoltage][, currentLimit]*) ###
+### enableCharger(*[,chargeVoltage][, currentLimit][,settings]*) ###
```
Comment (betzrhodes) [inline README.md]: the first 2 parameters should update to required parameters

File: README.md [hunk 9]
```diff
@@ -41,26 +42,15 @@ Enables and configures the battery charger with settings to perform a charging c
...
 
-### enableCharging() ###
+##### Settings Table #####
 
-Enables the device to automatically perform a charging cycle when a battery is connected and an input source is available. Charging is enabled by default.
-
-#### Parameters ####
+| Charging Fault | Value | Description |
+| --- | --- | --- |
+| *chargeCurrentOptimizer* | *1* |  Identify maximum power point without overload the input source
```
Comment (betzrhodes) [inline README.md]: I think this should be a boolean whether to enable/disable optimizer.

**Review by smittytone — COMMENTED** [review]
Only thing I'd note is that I would suggest changing _enableCharger()_ to _enable()_ and _disableCharger()_ to _disable()_. We'd normally use the shorter versions, and I think it's clear to anyone using this lib that they apply to a charger.

**Pull request merged**